File: KQXDrawer/KQXDrawer/LeftViewController.m
//
//  LeftViewController.m
//  KQXDrawer
//
//  Created by Kxx.xxQ 一生相伴 on 15/12/4.
//  Copyright © 2015年 asahi_kuang. All rights reserved.
//

#import "LeftViewController.h"
#import "AppDelegate.h"
#import "MenuPushViewController.h"

static NSString *const reuseIdentifier = @"identifier";
@interface LeftViewController ()<UITableViewDelegate>
@property (nonatomic, strong) UITableView *contenTable;
@property (nonatomic, strong) NSMutableArray *dataArray;
// 分离出去的dataSource
@property (nonatomic, strong) ArrayDataSource *dataSource;
@end

@implementation LeftViewController

#pragma mark - life cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [self createUI];
    [self setUpTableView];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
#pragma mark - lazy loading
- (NSMutableArray *)dataArray {
    if (!_dataArray) {
        _dataArray = [NSMutableArray array];
    }
    return _dataArray;
}

#pragma mark - mothods
- (void)createUI {
    UIImageView *leftBgImage = [[UIImageView alloc] initWithFrame:self.view.bounds];
    [leftBgImage setImage:[UIImage imageNamed:@"backgroundPic"]];
    [self.view addSubview:leftBgImage];
    
    UIImageView *userIcon = [[UIImageView alloc] initWithFrame:REACT_MAKE(65, 70, 100, 100)];
    [userIcon setImage:[UIImage imageNamed:@"user"]];
    [userIcon.layer setMasksToBounds:YES];
    [userIcon.layer setCornerRadius:50.f];
    [userIcon setTag:ICON_TAG];
    [leftBgImage addSubview:userIcon];
}

- (void)setUpTableView {
    _contenTable = [[UITableView alloc] initWithFrame:REACT_MAKE(0, 200, REACT_WIDTH(FRAME), REACT_HEIGHT(FRAME) - 200) style:UITableViewStylePlain];
    [_contenTable setDelegate:self];
    [_contenTable setSeparatorStyle:UITableViewCellSeparatorStyleSingleLineEtched];
    [_contenTable setBackgroundColor:[UIColor clearColor]];
    [_contenTable setTableFooterView:[UIView new]];
    [self.view addSubview:_contenTable];
    
    // 实现分离出去的dataSource
    cellConfigureBlock block = ^(UITableViewCell *cell, NSString *item) {
        [cell setBackgroundColor:[UIColor clearColor]];
        [cell.textLabel setTextColor:RGB(221, 55, 0)];
        [cell.textLabel setFont:BOLD(18.F)];
        [cell.textLabel setTextAlignment:NSTextAlignmentLeft];
        [cell setSelectionStyle:UITableViewCellSelectionStyleBlue];
        [cell.textLabel setText:item];
    };
    NSArray *contents = @[@"我的圈子", @"我的收藏", @"主题商城", @"AcFun", @"Bilibili", @"退出账号", @"设置"];
    [self.dataArray addObjectsFromArray:contents];
    _dataSource = [[ArrayDataSource alloc] initWithItems:self.dataArray withReuseIdentifier:reuseIdentifier withCellConfigureBlock:block isSystemCell:YES];
    [_contenTable setDataSource:_dataSource];
}

#pragma mark - UITableViewDelegate
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:NO];
    AppDelegate *deleg = [[UIApplication sharedApplication] delegate];
    [deleg.leftSlideVC closeLeftViewController];
    
    MenuPushViewController *mpVC = [[MenuPushViewController alloc] init];
    [mpVC setNavTitleStr:self.dataArray[indexPath.row]];
    [deleg.mainNav pushViewController:mpVC animated:YES];
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 50.f;
}
/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: KXXDrawer_Swift/KXXDrawer_Swift/LeftViewController.swift
//
//  LeftViewController.swift
//  KXXDrawer_Swift
//
//  Created by Kxx.xxQ 一生相伴 on 15/12/4.
//  Copyright © 2015年 asahi_kuang. All rights reserved.
//

import UIKit

let reuseIdentifierr = "identifier"
class LeftViewController: UIViewController, UITableViewDelegate {
    var tableView: UITableView?
    var dataArray: Array<AnyObject>?
    var dataSource: ArrayDataSource?
    // MARK: - life cycle
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        self.createUI()
        self.makeUpTableView()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - Create UI
    func createUI() {
        self.view.backgroundColor = UIColor.whiteColor()
        let bgImage = UIImageView(frame: CGRectMake(0, 0, CGRectGetWidth(self.view.frame), CGRectGetHeight(self.view.frame)))
        bgImage.image = UIImage.init(imageLiteral: "backgroundPic")
        self.view.addSubview(bgImage)
        
        let userIcon = UIImageView(frame: CGRectMake(65, 70, 100, 100))
        userIcon.image = UIImage.init(imageLiteral: "user")
        userIcon.layer.masksToBounds = true
        userIcon.layer.cornerRadius  = 50
        userIcon.tag = 1991
        bgImage.addSubview(userIcon)
    }
    
    // MARK: - makeUpTableView
    func makeUpTableView() {
        //REACT_MAKE(0, 180, REACT_WIDTH(FRAME) - SCREEN_WIDTH/2+65, REACT_HEIGHT(FRAME) - 150)
        self.tableView = UITableView(frame: CGRectMake(0, 180, CGRectGetWidth(self.view.frame) - CGRectGetWidth(self.view.frame)/2 + 65, CGRectGetHeight(self.view.frame)), style: UITableViewStyle.Plain)
        let tbView = UIView.init()
        self.tableView!.tableFooterView = tbView
        self.tableView!.delegate = self
        self.tableView!.backgroundColor = UIColor.clearColor()
        self.tableView!.separatorStyle = UITableViewCellSeparatorStyle.None
        
        let block: cellConfigureBlock? = {
            cell, item in

            let myCell = (cell as! UITableViewCell)
            let title = (item as! String)
            
            myCell.textLabel?.text = title
            myCell.textLabel?.font = UIFont.boldSystemFontOfSize(18)
            myCell.textLabel?.textColor = UIColor(red: 43/255.0, green: 60/255.0, blue: 47/255.0, alpha: 1)
            myCell.backgroundColor = UIColor.clearColor()
        }
        
        
        self.dataArray = ["我的主页", "我的圈子", "AcFun", "Bilibili", "更换主题", "设置", "退出"]
        
        self.dataSource = ArrayDataSource(items: self.dataArray!, reuseIdentifier: reuseIdentifierr, cellBlock: block, isSys: true)
        self.tableView!.dataSource = self.dataSource
        self.view.addSubview(self.tableView!)
    }

    // MARK: - UITableViewDelegate
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        tableView.deselectRowAtIndexPath(indexPath, animated: false)
        let appdelegate = UIApplication.sharedApplication().delegate as! AppDelegate
        appdelegate.leftVc?.closeLeftSlideViewController()
        
        let pushVC = NewPushViewController()
        pushVC.navTitle = (self.dataArray![indexPath.row] as! String)
        appdelegate.mainNav?.pushViewController(pushVC, animated: true)
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
